src/jsxc.lib.xmpp.mam.js

```diff
@@ -27,14 +27,111 @@
  *
  */
 
-/*jsxc.xmpp.mam = {
+jsxc.xmpp.mam = {
   conn: jsxc.xmpp.conn
 };
 
 jsxc.xmpp.mam.init = function(){
-  var self = jsxc.xmpp.mam;
-  self.getHistory(userid, count, before);
-};*/
+  var self = jsxc.xmpp.conn.mam;
+  console.log(self);
+
+  //self has now the object of the strophe.mam
+  // make a call to the get history function.
+
+
+  //self.getHistory(userid, count, before);
+};
+
+
+jsxc.xmpp.mam.getHistory = function(bid) {
+  /*
+  To include the whole functionality of strophe-plugin we used self
+  */
+  var self = jsxc.xmpp.conn.mam;
+  /*
+  ownjid is the jid of the person who is logged in
+  */
+  var ownjid = jsxc.xmpp.conn.jid;
+
+  // Now we are converting the jid into bid
+  var ownbid = jsxc.jidToBid(ownjid);
+
+
+  // We are calling the strophe plugin as specified by them
+  self.query(ownbid, {
+    with: bid, 
+    before: "", 
+    onMessage: function(message) {
+      /*
+      The message looks like this:
+      <message xmlns="jabber:client" from="ownbid" to="bid">
+        <result xmlns="urn:xmpp:mam:0" id="1485155123866746">
+          <forwarded xmlns="urn:xmpp:forward:0">
+            <message xmlns="jabber:client" from="jid from which message is sent" to="bid who recived the message" type="chat" id="1485141093744:msg">
+              <body>gdhjcb</body>
+              <request xmlns="urn:xmpp:receipts"></request>
+              <delay xmlns="urn:xmpp:delay" from="serverAddress" stamp="2017-01-23T03:11:43.446Z">Offline Storage</delay>
+            </message>
+            <delay xmlns="urn:xmpp:delay" from="serverAddress" stamp="2017-01-23T07:05:23.867Z"></delay>
+          </forwarded>
+        </result>
+        <no-store xmlns="urn:xmpp:hints"></no-store>
+      </message>
+      
+      */
+      var direction;
+
+      /*
+      Check the direction of the message i.e. In which direction messages are exchanged.
+      */
+      if(bid === $(message).find("forwarded message").attr("to")){
```

Comment:
This is always equal to "<forwarded xmlns="urn:xmpp:forward:0">". What do we need to check here?